File: 365/water-and-jug-problem.py
class Solution:
    def canMeasureWater(self, x, y, z):
        """
        :type x: int
        :type y: int
        :type z: int
        :rtype: bool
        """
        return z == 0 or (x + y >= z and z % self.gcd(x, y) == 0)

    def gcd(self, x, y):
        if y == 0:
            return x
        return self.gcd(y, x % y)

File: 365/water-and-jug-problem.go
package main

import "fmt"

func canMeasureWater(x int, y int, z int) bool {
	return z == 0 || (x+y >= z && z%gcd(x, y) == 0)
}

func gcd(x int, y int) int {
	if y == 0 {
		return x
	}
	return gcd(y, x%y)
}

func main() {
	fmt.Println(canMeasureWater(3, 5, 4))
}
